File: Lucidity/LucidityControl.vb
Imports System.Xml
Imports System.IO

Public Class LucidityControl

  Public Event LucidFolderSelected(ByVal Category As String)

  Private Sub lstLucdity_MouseMove(ByVal sender As System.Object, ByVal e As System.Windows.Forms.MouseEventArgs) Handles lstLucidity.MouseMove
    If Not lstLucidity.GetItemAt(e.X, e.Y) Is Nothing Then
      lstLucidity.Cursor = Cursors.Hand
    Else
      lstLucidity.Cursor = Cursors.Default
    End If
  End Sub

  Private Sub lstLucdity_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles lstLucidity.SelectedIndexChanged
    If lstLucidity.SelectedItems.Count = 0 Then Return
    RaiseEvent LucidFolderSelected(lstLucidity.SelectedItems(0).Text)
  End Sub

End Class

File: Lucidity/LucidityControl.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.Xml;
using System.IO;
namespace lucidcode.LightenedDream.Lucidity
{

	public partial class LucidityControl
	{

		public event LucidFolderSelectedEventHandler LucidFolderSelected;
		public delegate void LucidFolderSelectedEventHandler(string Category);

		private void lstLucdity_MouseMove(System.Object sender, System.Windows.Forms.MouseEventArgs e)
		{
			if ((lstLucidity.GetItemAt(e.X, e.Y) != null)) {
				lstLucidity.Cursor = Cursors.Hand;
			} else {
				lstLucidity.Cursor = Cursors.Default;
			}
		}

		private void lstLucdity_SelectedIndexChanged(System.Object sender, System.EventArgs e)
		{
			if (lstLucidity.SelectedItems.Count == 0)
				return;
			if (LucidFolderSelected != null) {
				LucidFolderSelected(lstLucidity.SelectedItems[0].Text);
			}
		}
		public LucidityControl()
		{
			InitializeComponent();
		}

	}
}
